### ISSUE TITLE
creation of devel projects for slfo

### ISSUE THREAD

**Issue by celan** [aplanas/openSUSE-git#63]
Group all SLFO packages into devel projects and create them on [src.suse.de/link](http://src.suse.de/link) them to IBS hosted projects

**Comment by dirkmueller** [comment]
I think these are the options while keeping `openSUSE:Factory` and Factory First in mind:

* create src organisations that are named after the most specific component of the current devel project names, e.g. usually the last component:

```
Base:System -> src.opensuse.org/system
devel:languages:python -> src.opensuse.org/python
devel:languages:python:Factory -> src.opensuse.org/python
devel:libraries:c_c++ -> src.opensuse.org/c_c++ or src.opensuse.org/libraries
```

The downside is that team lists are org specific, so we can not convert them into shared team lists natively. could be done with an extra bot though.

**Comment by AdamMajer** [comment]
I think this will have to be done manually, on a case-by-case basis. Some devel projects have structures in their hierarchy. Some have ideas. Some have chaos.

For importing a devel project, there's https://src.opensuse.org/adamm/autogits/src/branch/main/devel-importer . It doesn't import maintainership info yet.

**Comment by mcepl** [comment]
> devel:languages:python -> src.opensuse.org/python
> devel:languages:python:Factory -> src.opensuse.org/python

I will not lean on sword for d:l:p:Factory (even though, it is useful, so that our experiments with interpreter don’t break d:l:p), but I will require at least `python_numeric` and `python_cloud` (if @glaubitz agrees, perhaps even more detailed organization).

**Comment by dirkmueller** [comment]
> but I will require at least python_numeric and python_cloud (if @glaubitz agrees, perhaps even more detailed organization).

why? it's just how sources are organized, not how sources are built or how they are maintained (the maintainership is done via teams defined in the org). having one org means you can define a "numeric-team" and a "cloud-team" and associate them on a repository level. which means even if a package moves between one or the other devel project, they don't have to move sources. which is a huge benefit imho.

**Comment by dirkmueller** [comment]
> I think this will have to be done manually, on a case-by-case basis. 

if we wait for 12000 packages to be manually sorted in organisations, we're still waiting for this in 10 years from now. we need to start with a good default and allow people to move..

> Some devel projects have structures in their hierarchy. Some have ideas. Some have chaos.

sure, and we can do some cleanup beforehand. but the fact that the devel projects as they are today work for the people today is not unimportant.

**Comment by AdamMajer** [comment]
As far as I can tell, anyone can setup any organization name in Gitea. There should be not much naming restrictions.

By case-by-case basis, I meant that some OBS projects could be merged into one (eg. python with different branches representing different setups). Other OBS projects will not be merged or maybe not even be migrated because they are no longer used. It will need consultation with project maintainers how they want to proceed.

We are not going to be re-sorting packages between projects. Importing Factory devel projects will be priority.

**Comment by dirkmueller** [comment]
> As far as I can tell, anyone can setup any organization name in Gitea. There should be not much naming restrictions.

That is true, but do we really want to let everyone create their own organisation using their own naming pattern? how accessible is that for new contributors?

> By case-by-case basis, I meant that some OBS projects could be merged into one (eg. python with different branches representing different setups). Other OBS projects will not be merged or maybe not even be migrated because they are no longer used. It will need consultation with project maintainers how they want to proceed.

Sure, that can be done, based on a proposal that needs to be written up before approaching people for veto/confirmation?


> We are not going to be re-sorting packages between projects. Importing Factory devel projects will be priority.

but you need to import those project gits *somewhere*, so we need the *orgs* to be able to host them?

**Comment by AdamMajer** [comment]
> That is true, but do we really want to let everyone create their own organisation using their own naming pattern? how accessible is that for new contributors?

We'll need functionality equivalent to `osc develproject openSUSE:Factory/cockpit` . Maybe enforcing a README.md with such information in factory packages would be an answer here.

**Comment by mcepl** [comment]
> why? it's just how sources are organized, not how sources are built or how they are maintained (the maintainership is done via teams defined in the org). having one org means you can define a "numeric-team" and a "cloud-team" and associate them on a repository level.

Well, the question will be whether there is any relationship between OBS devel projects and these organizations in Gitea. Will be there? If not, then “python-team” Gitea organization makes sense.

**Comment by AdamMajer** [comment]
Gitea Orgs will have a 1:1 mapping to devel projects in OBS.

You are welcome to setup additional Gitea Organizations as you want and need. Please clean them up once they are unnecessary.

**Comment by mmaslano** [comment]
I guess this is sort of done. @AdamMajer is transfering devel projects. There exist pool project to resolve the interim.
